### DISCUSSION TITLE
Converting MRF from XML to JSON for Upload

### DISCUSSION THREAD

**Discussion by jrdanielpr** [CMSgov/hospital-price-transparency#35] [Q&A] [answered]
Hello. We would like to know if it is permissible to create our machine readable file in Excel, following the exact format provided by CMS, and then convert that .XML file to .JSON so that the .JSON version can be uploaded to GitHub. We note that, during internal testing, while we are following CMS’s exact formatting guidelines to create the .XML file, there are some minor discrepancies in formatting once we convert the file to .JSON. It is possible this discrepancy is attributable to the third-party software program we used to read the .JSON file. Would the hospital be required to manually correct the file output in order to be deemed in compliance? Your response is appreciated as we await rollout of the Validator 2.0 tool this Spring 2024. Thank you.

**Comment by RidenourK** [comment, marked as answer]
@jrdanielpr. Thank you for your question. 

As finalized in the CY2024 OPPS/ASC final rule, beginning July 1, 2024, all hospitals must adopt a CMS template and encode its standard charge information in it **in the form and manner** finalized in the [Hospital Price Transparency Data Dictionary GitHub repository](https://github.com/CMSgov/hospital-price-transparency).
 
We suggest you upload your file to the Machine-Readable File Validator 2.0 when it is available on the [Hospital Price Transparency Tools](https://cmsgov.github.io/hpt-tool/) page in Spring of 2024. The updated validator will test machine-readable files against the new required CMS template layout and data specifications that hospitals must conform to by July 1, 2024.
